toolchain/mfc/case_validator.py

```
@@ -1673,9 +1885,34 @@ def validate(self, stage: str = 'simulation'):
 
+    def _format_errors(self) -> str:
+        """Format errors with enhanced context and suggestions."""
+        lines = ["[bold red]Case parameter constraint violations:[/bold red]\n"]
+
+        for i, err in enumerate(self.errors, 1):
+            lines.append(f"  [bold]{i}.[/bold] {err}")
+
+            # Add helpful hints for common errors
+            err_lower = err.lower()
+            if "must be positive" in err_lower or "must be set" in err_lower:
+                lines.append(f"     [dim]Check that this required parameter is defined in your case file[/dim]")
```

Ruff (0.14.14)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)